### PR TITLE
refactor(datepicker): implement one time bindings

### PR DESCRIPTION
**Pull request by RobJacobs**
Reducing the number of watches created for the 
datepicker by implementing one time binding
on the datepicker templates. Removing aria-disabled
attribute as the ngAria module will add that
to the button with ng-disabled.

### CODE REVIEW

File: template/datepicker/day.html [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-      <th colspan="{{5 + showWeeks}}"><button id="{{uniqueId}}-title" role="heading" aria-live="assertive" aria-atomic="true" type="button" class="btn btn-default btn-sm" ng-click="toggleMode()" ng-disabled="datepickerMode === maxMode" tabindex="-1" style="width:100%;"><strong>{{title}}</strong></button></th>
+      <th colspan="{{5 + showWeeks}}"><button id="{{::uniqueId}}-title" role="heading" aria-live="assertive" aria-atomic="true" type="button" class="btn btn-default btn-sm" ng-click="toggleMode()" ng-disabled="datepickerMode === maxMode" tabindex="-1" style="width:100%;"><strong>{{title}}</strong></button></th>
       <th><button type="button" class="btn btn-default btn-sm pull-right" ng-click="move(1)" tabindex="-1"><i class="glyphicon glyphicon-chevron-right"></i></button></th>
     </tr>
     <tr>
-      <th ng-show="showWeeks" class="text-center"></th>
-      <th ng-repeat="label in labels track by $index" class="text-center"><small aria-label="{{label.full}}">{{label.abbr}}</small></th>
+      <th ng-show="::showWeeks" class="text-center"></th>
```
Comment (Foxandxss) [inline template/datepicker/day.html]: Not sure if one.time on ng-show is gonna work.

Comment (RobJacobs) [inline template/datepicker/day.html]: I should be using ng-if, but then tests fail.

**Comment by wesleycho** [comment]
Is removing aria-disabled a good idea? Users of ngAria would want that, no?

**Comment by RobJacobs** [comment]
Partially fixes #2613, should compliment #1915

**Comment by realityking** [comment]
From my understanding of the [ARIA spec](http://www.w3.org/WAI/PF/aria/roles#gridcell), the `aria-disabled` should stay as this indicated that the griddle is disabled, not just the button within it.

BTW, for buttons, ngAria is not needed. All modern browsers map the accessible information correctly for disabled button. Unfortunately ngAria will add unnecessary attributes to the DOM anyway.

**Comment by RobJacobs** [comment]
@wesleycho @realityking I should have done a better job of explaining why I think the aria-disabled binding should be removed on the td element.  Here is a [plunk](http://plnkr.co/edit/nhQOpKnWjeYvzElzuxPb?p=preview) with the latest build from master, Angular 1.3, and ngAria loaded.  You will notice the rendered html for the day/month/year picker buttons when disabled looks like:

    <td aria-disabled="true" ng-repeat="dt in row track by dt.date">
      <button tabindex="-1" disabled="disabled" aria-disabled="true"

Using ng-disabled directive on the button element will cause the ngAria service to add the aria-disabled attribute for you.  I don't think it's necessary to have the aria-disabled attribute on both the td and button elements.  

The rendered html for an enabled button doesn't have the disabled or aria-disabled attributes at all where the enabled td element aria-disabled attribute is rendered as aria-disabled="false".  That leads me to believe the correct implementation for using aria-disabled is to remove or add the attribute versus setting the value.

**Comment by realityking** [comment]
I've spent some time with the ARIA spec, but I'm far from an expert.

Still there are a few things worth clarifying:

1. Unlike the `disabled` attribute, `aria-disabled` (and other aria states) is not a [boolean](http://stackoverflow.com/questions/4139786/what-does-it-mean-in-html-5-when-an-attribute-is-a-boolean-attribute) attribute whose presence or absence determines what's going on, the value actually matters. (Otherwise we'd write `aria-hidden="aria-hidden"`). It's a confusing mess, but that's what you get from an evolving system.

2. In my opinion, the `aria-disabled`attribute is correct, because it describes the state of the gridcell, not that of the button within it.

The ARIA Best Practices document has a section on datepicker [date picker](http://www.w3.org/TR/wai-aria-practices/#datepicker), it doesn't mention `aria-disabled` explicitly but the linked implementation in Dojo does use it.

**Comment by chrisirhc** [comment]
I like the change to remove aria-disabled, and leave it to ngAria to handle that for us since we're moving to AngularJS 1.3 anyway.

**Comment by hamfastgamgee** [comment]
I've been running with a closely-related variant of this and #1915 in my local fork for a long time, so I'm going to add a couple of suggestions to the patch if you don't mind. :)

(For reference, we have one data entry page that has probably 50 datepickers on it.  Without this and #1915, let's just say that page would not be loading in under 20 seconds...)

**Comment by RobJacobs** [comment]
@hamfastgamgee I decided to err on the side of caution, I can't imagine why someone would want to change the show weeks or start day attributes while the datepicker is visible, but you never now.

**Comment by hamfastgamgee** [comment]
Fair enough.  Might warrant further discussion as to whether those are actually desired, though, since at least in the start date case it's one extra watcher per day visible.  (The showWeeks stuff is one per picker, so not as big of an issue.)

**Comment by wesleycho** [comment]
@RobJacobs what's the state of the PR? Since we released 0.13.0, we are in the position to merge this if this is ready.

**Comment by RobJacobs** [comment]
@wesleycho It's good to go as far as I'm concerned.  Rebased and tests still passing.

**Comment by wesleycho** [comment]
Alright, I'll review it tonight

**Comment by BobbieBarker** [comment]
I approve

File: src/datepicker/test/datepicker.spec.js [hunk 2]
```diff
@@ -59,10 +59,10 @@ describe('datepicker directive', function () {
     return element.find('thead').find('tr').eq(1);
   }
 
-  function getLabels() {
+  function getLabels( dayMode ) {
```
Comment (wesleycho) [inline src/datepicker/test/datepicker.spec.js]: Remove spaces around `dayMode`

**Comment by wesleycho** [comment]
LGTM minus one code style comment.

**Pull request closed without merging**